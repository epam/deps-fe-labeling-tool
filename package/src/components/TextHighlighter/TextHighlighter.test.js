import React from 'react'
import { shallow } from 'enzyme'
import { TextHighlighter } from '.'

describe('TextHighlighter', () => {
  let defaultProps
  let wrapper

  beforeEach(() => {
    defaultProps = {
      text: 'mock Text',
      searchTerm: 'm'
    }

    wrapper = shallow(<TextHighlighter {...defaultProps} />)
  })

  it('should render layout correctly in case there is a match in text', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render layout correctly in case there is no match in text', () => {
    wrapper.setProps({
      ...defaultProps,
      searchTerm: 'hey'
    })

    expect(wrapper).toMatchSnapshot()
  })
})
