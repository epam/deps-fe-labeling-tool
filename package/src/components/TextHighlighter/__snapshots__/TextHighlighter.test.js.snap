// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextHighlighter should render layout correctly in case there is a match in text 1`] = `
<span>
  <TextHighlighterstyles__Mark
    key="0"
  >
    m
  </TextHighlighterstyles__Mark>
  ock Text
</span>
`;

exports[`TextHighlighter should render layout correctly in case there is no match in text 1`] = `
<span>
  mock Text
</span>
`;
